src/conversion.c

```diff
@@ -341,26 +353,49 @@ int igraph_get_edgelist(const igraph_t *graph, igraph_vector_t *res, igraph_bool
 
 int igraph_to_directed(igraph_t *graph,
                        igraph_to_directed_t mode) {
-
-    if (mode != IGRAPH_TO_DIRECTED_ARBITRARY &&
-        mode != IGRAPH_TO_DIRECTED_MUTUAL) {
-        IGRAPH_ERROR("Cannot direct graph, invalid mode", IGRAPH_EINVAL);
-    }
+    long int no_of_edges = igraph_ecount(graph);
+    long int no_of_nodes = igraph_vcount(graph);
 
     if (igraph_is_directed(graph)) {
-        return 0;
-    }
-
-    if (mode == IGRAPH_TO_DIRECTED_ARBITRARY) {
-
+        return IGRAPH_SUCCESS;
+    }   
+
+    switch (mode) {
+    case IGRAPH_TO_DIRECTED_ARBITRARY:
+    case IGRAPH_TO_DIRECTED_RANDOM:
+    case IGRAPH_TO_DIRECTED_ACYCLIC:
+      {
         igraph_t newgraph;
         igraph_vector_t edges;
-        long int no_of_edges = igraph_ecount(graph);
-        long int no_of_nodes = igraph_vcount(graph);
         long int size = no_of_edges * 2;
         IGRAPH_VECTOR_INIT_FINALLY(&edges, size);
         IGRAPH_CHECK(igraph_get_edgelist(graph, &edges, 0));
 
+        if (mode == IGRAPH_TO_DIRECTED_ACYCLIC) {
+            long int i;
```

Comment:
Basically all branches seem to have an `i` counter variable. Can we lift the declaration to the top-level?

Reply:
OK, done that. Lots of variables could be shared between the two branches of the `switch` as well (originally an if-else). I left those mostly as they were before, and only lifted the vertex and edge-count variables. The reason is that if we add more methods in the future, not everything may be shared between them.